components/configuration-mgt/org.wso2.carbon.identity.configuration.mgt.core/src/main/java/org/wso2/carbon/identity/configuration/mgt/core/dao/impl/ConfigurationDAOImpl.java

```diff
@@ -652,9 +652,15 @@ public void addResource(Resource resource) throws ConfigurationManagementExcepti
                         preparedStatement -> {
                             int initialParameterIndex = 1;
                             preparedStatement.setString(initialParameterIndex, resource.getResourceId());
-                            preparedStatement.setInt(++initialParameterIndex,
-                                    PrivilegedCarbonContext.getThreadLocalCarbonContext()
-                                            .getTenantId());
+                            int tenantId;
+                            String tenantDomain = IdentityTenantUtil.getTenantDomainFromContext();
+                            if (StringUtils.isBlank(tenantDomain)) {
+                                tenantId = PrivilegedCarbonContext.getThreadLocalCarbonContext()
+                                        .getTenantId();
+                            } else {
+                                tenantId = IdentityTenantUtil.getTenantId(tenantDomain);
+                            }
+                            preparedStatement.setInt(++initialParameterIndex, tenantId);
```

Comment:
Is this needed. Isn't the expectation of the dao is to read tenant from PrivilegedCarbonContext.
In that case, we can have a tenant flow initiated from the service layer right

Reply:
It seems that correct tenant domain is retrieved from PrivilegedCarbonContext even though we don't have a tenant flow initiated from the service layer. So shall I keep the old code as it is?

Reply:
+1